vlib/v/checker/tests/for_match_err.vv

```diff
@@ -0,0 +1,15 @@
+fn main() {
+  mut a := 2
+  for match a {
+		2 {
+			println('a == 2')
+			a = 0
+		}
+		0 {
+			println('a == 0')	
+		}
+		else {
+			println('unexpected branch')
+		}
```

Comment:
This is a match statement, not a match expression so it shouldn't compile because `for match` and `if match` need it to produce a result. If match is an expression that produces a bool I don't see why we should disallow it.

Reply:
The reasoning is making code more readable. `for match { ... } { } ` is very confusing and wouldn't be used often anyway.

Reply:
It could be a v vet warning then. (An error is ok but then why not make all v vet warnings into v errors).

Reply:
@ntrel `vet` is used to analyze code and give suggestions about it. I feel `for match` and `if match` ain't sensible at all. Also Rust doesn't allow the `match` in these cases. Thus, it should be an error.

Reply:
On reflection I think making it an error is a good choice as it helps to simplify cgen (as well as for readability).